server/autocomplete_search.go

```diff
@@ -57,30 +55,28 @@ func (p *Plugin) httpGetSearchUsers(w http.ResponseWriter, r *http.Request) (int
 
 	client, _, _, err := p.getClient(types.ID(instanceID), types.ID(mattermostUserID))
 	if err != nil {
-		return http.StatusInternalServerError, err
+		return respondErr(w, http.StatusInternalServerError, err)
 	}
 
 	// Get list of assignable users
 	jiraUsers, err := client.SearchUsersAssignableInProject(projectKey, userSearch, 10)
 	if StatusCode(err) == 401 {
-		return http.StatusInternalServerError, err
+		return respondErr(w, http.StatusInternalServerError, err)
 	}
 
 	if jiraUsers == nil {
-		return http.StatusInternalServerError, errors.New("failed to return any results")
+		return respondErr(w, http.StatusInternalServerError, errors.New("failed to return any results"))
 	}
 
 	bb, err := json.Marshal(jiraUsers)
 	if err != nil {
-		return http.StatusInternalServerError,
-			errors.WithMessage(err, "failed to marshal response")
+		return respondErr(w, http.StatusInternalServerError, errors.WithMessage(err, "failed to marshal response"))
 	}
 
 	w.Header().Set("Content-Type", "application/json")
 	_, err = w.Write(bb)
 	if err != nil {
-		return http.StatusInternalServerError,
-			errors.WithMessage(err, "failed to write response")
+		return http.StatusInternalServerError, errors.WithMessage(err, "failed to write response")
```

Comment:
I notice we're not calling `respondErr` here. I think that's intentional since this error is coming from a failure to write a response

Reply:
@mickmister We were not calling `respondErr` anywhere in the function. I think we should as the user will not see any data in the dropdown in this case as well. Please let me know your thoughts on this.